lib/util/probe.js

```diff
@@ -60,8 +64,50 @@ function observeKafkaStats(msg) {
     kafkaMetrics.observe(JSON.parse(msg.message));
 }
 
+/**
+ * Get probe config will pull the configuration for the probe server based on
+ * the provided site name. If siteNames is empty, it returns the global probe server config
+ * only if it's a single object.
+ *
+ * @param {Object} queueProcessorConfig - Configuration of the queue processor that
+ *      holds the probe server configs for all sites
+ * @param {Array<String>} siteNames - List of site names (should contain at most one element)
+ * @returns {Object|undefined} Config for site or global config, undefined if no match found or invalid config
+ */
+function getProbeConfig(queueProcessorConfig, siteNames) {
+    const logger = new Logger('Backbeat:Probe:getProbeConfig');
+
+    if (siteNames.length === 0) {
+        if (!Array.isArray(queueProcessorConfig.probeServer)) {
+            return queueProcessorConfig.probeServer;
+        }
+
+        logger.error('Configuration set for specific sites, but no site provided to the process', {
+            siteNames,
+            queueProcessorConfig,
+        });
+        return undefined;
+    }
+
+    if (Array.isArray(queueProcessorConfig.probeServer)) {
+        if (siteNames.length !== 1) {
+            logger.error('Process configured for more than one site', {
+                siteNames,
+                queueProcessorConfig,
+            });
+            return undefined;
+        }
+
+        const siteConfig = queueProcessorConfig.probeServer.find(config => config.site === siteNames[0]);
+        return siteConfig || undefined;
+    }
+
+    return undefined;
```

Comment:
Shouldn't this return `queueProcessorConfig.probeServer` ? We fall in this case when having multiple sites and only one probeserver config, which is how it's deployed in Artesca.